### PR TITLE
[stable10] Accept user share stable10

### PR DESCRIPTION
**Pull request by jvillafanez**
## Description
Backport of https://github.com/owncloud/core/pull/30106 to stable10

## Related Issue
https://github.com/owncloud/core/issues/19153

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

@owncloud/qa please verify the .drone.yml configuration to recheck all the tests we want to run are there. There were some conflicts there so I've mostly copied whatever there is in the master branch.

### CODE REVIEW

**Comment by PVince81** [comment]
strange failure with swift and postgres run:
```
PHP Fatal error: Cannot use Test\TestCase as TestCase because the name is already in use in /drone/src/apps/files_sharing/tests/HooksTest.php on line 30
```

Not sure if related at all

**Comment by jvillafanez** [comment]
I'm not sure if we should move the fix for the tests to master. The tests aren't complaining about it though.

**Comment by PVince81** [comment]
@patrickjahns please review the extra commit https://github.com/owncloud/core/pull/31613/commits/75f77a622469ed285e3195939b1e0dba01cc3d34.

File: .drone.yml [hunk 1]
```diff
@@ -136,6 +136,26 @@ pipeline:
       matrix:
         INSTALL_SERVER: true
 
+  setup-caldav-tests:
```
Comment (patrickjahns) [inline .drone.yml]: this is not related to this pr ?

Comment (jvillafanez) [reply .drone.yml]: It seems so. It seems it was added with https://github.com/owncloud/core/pull/31613/commits/484f89c1912f66a43bdb0fbbea0af6482afd0e82 . All the changes were cherry-picked except the last one (to fix breaking things), so not sure why this was needed in the master PR.

Feel free to remove it.

**Comment by phil-davis** [comment]
Rebased and added commit from #31661 to resolve ``run.sh`` conflict.

**Comment by PVince81** [comment]
It's green.

Any update on https://github.com/owncloud/core/pull/31613#discussion_r193309072 ? Is this acceptable ?

File: .drone.yml (hunk 1 above)
Comment (patrickjahns) [reply .drone.yml]: please remove the step - it is unrelated to this PR

**Review by patrickjahns — CHANGES_REQUESTED** [review]
Please remove unnecessary .drone step

**Comment by phil-davis** [comment]
I will have a look at the drone conflict... and rebase/push in the next few minutes.

**Comment by phil-davis** [comment]
Conflict resolved.

**Comment by PVince81** [comment]
Green again, no more conflict. Merge ?

**Comment by phil-davis** [comment]
Rebased and added the 4 commits from #31665

**Review by phil-davis — APPROVED** [review]
Looks OK now to me. I guess that @patrickjahns review has been addressed? Because I do not see a problem with ``.drone.yml``any more.

**Review by patrickjahns — APPROVED** [review]

**Pull request merged**